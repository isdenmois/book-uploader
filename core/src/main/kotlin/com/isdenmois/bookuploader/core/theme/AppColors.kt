package com.isdenmois.bookuploader.core.theme

import androidx.compose.runtime.Stable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.compose.ui.graphics.Color

@Stable
class AppColors(
    background: Color,
    searchBackground: Color,
    uploadBackground: Color,
    profileBackground: Color,
    tabsBackground: Color,
    secondaryBackground: Color,
    progressBackground: Color,
    modalBackground: Color,

    searchText: Color,
    uploadText: Color,
    profileText: Color,
    tabText: Color,

    searchSelected: Color,
    uploadSelected: Color,
    profileSelected: Color,

    primary: Color,
    invertedText: Color,
    secondary: Color,
    search: Color,
    input: Color,
    deleteButton: Color,
    success: Color,
    error: Color,
    isLight: Boolean
) {
    var background by mutableStateOf(background)
        private set
    var searchBackground by mutableStateOf(searchBackground)
        private set
    var uploadBackground by mutableStateOf(uploadBackground)
        private set
    var profileBackground by mutableStateOf(profileBackground)
        private set
    var tabsBackground by mutableStateOf(tabsBackground)
        private set
    var secondaryBackground by mutableStateOf(secondaryBackground)
        private set
    var progressBackground by mutableStateOf(progressBackground)
        private set
    var modalBackground by mutableStateOf(modalBackground)
        private set

    var searchText by mutableStateOf(searchText)
        private set
    var uploadText by mutableStateOf(uploadText)
        private set
    var profileText by mutableStateOf(profileText)
        private set
    var tabText by mutableStateOf(tabText)
        private set

    var searchSelected by mutableStateOf(searchSelected)
        private set
    var uploadSelected by mutableStateOf(uploadSelected)
        private set
    var profileSelected by mutableStateOf(profileSelected)
        private set

    var primary by mutableStateOf(primary)
        private set
    var invertedText by mutableStateOf(invertedText)
        private set
    var secondary by mutableStateOf(secondary)
        private set
    var search by mutableStateOf(search)
        private set
    var input by mutableStateOf(input)
        private set
    var deleteButton by mutableStateOf(deleteButton)
        private set
    var success by mutableStateOf(success)
        private set
    var error by mutableStateOf(error)
        private set
    var isLight by mutableStateOf(isLight)
        private set

    fun update(other: AppColors) {
        background = other.background
        searchBackground = other.searchBackground
        uploadBackground = other.uploadBackground
        profileBackground = other.profileBackground
        tabsBackground = other.tabsBackground
        secondaryBackground = other.secondaryBackground
        progressBackground = other.progressBackground
        modalBackground = other.modalBackground

        searchText = other.searchText
        uploadText = other.uploadText
        profileText = other.profileText
        tabText = other.tabText

        searchSelected = other.searchSelected
        uploadSelected = other.uploadSelected
        profileSelected = other.profileSelected

        primary = other.primary
        invertedText = other.invertedText
        secondary = other.secondary
        search = other.search
        input = other.input
        deleteButton = other.deleteButton
        success = other.success
        error = other.error
        isLight = other.isLight
    }

    fun copy(): AppColors = AppColors(
        background = background,
        searchBackground = searchBackground,
        uploadBackground = uploadBackground,
        profileBackground = profileBackground,
        tabsBackground = tabsBackground,
        secondaryBackground = secondaryBackground,
        progressBackground = progressBackground,
        modalBackground = modalBackground,

        searchText = searchText,
        uploadText = uploadText,
        profileText = profileText,
        tabText = tabText,

        searchSelected = searchSelected,
        uploadSelected = uploadSelected,
        profileSelected = profileSelected,

        primary = primary,
        invertedText = invertedText,
        secondary = secondary,
        search = search,
        input = input,
        deleteButton = deleteButton,
        success = success,
        error = error,
        isLight = isLight,
    )
}
