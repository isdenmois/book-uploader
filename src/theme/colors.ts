export const primary = '#000';
export const secondary = '#eee';
export const invert = '#fff';

export const green = '#8BC34A';
export const red = '#f44336';
export const darkBlue = '#007ac1';
export const blue = '#2196f3';
export const lightBlue = '#67daff';

export const Background = '#F9FAFE';
export const SearchBackground = '#B8D5F9';
export const UploadBackground = '#C2B9F9';
export const ProfileBackground = '#FFAD99';
export const TabsBackground = 'white';

export const SearchText = '#092D5D';
export const UploadText = '#15085E';
export const ProfileText = '#661400';
export const TabText = '#121417';

export const SearchSelected = '#5396EE';
export const UploadSelected = '#6851F0';
export const ProfileSelected = '#FF6842';

export const Secondary = '#787D87';
export const Search = '#8C93A3';
export const Input = 'white';
export const DeleteButton = '#9BA8CA';
export const Success = '#48B755';
export const Error = '#FD4463';
