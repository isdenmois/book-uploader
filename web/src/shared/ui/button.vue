<script setup lang="ts">
defineProps<{
  disabled?: boolean
  type?: string
}>()
</script>
<template>
  <button :disabled="disabled" :type="type">
    <slot />
  </button>
</template>

<style scoped>
button {
  background-color: var(--profileSelected);
  border-radius: 10px;
  color: var(--invertedText);
  cursor: pointer;
  font-weight: bold;
  padding: 5px 20px;
}

button:disabled {
  opacity: 0.4;
}
</style>
