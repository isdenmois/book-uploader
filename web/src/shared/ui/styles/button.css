.button {
  background-color: var(--profileSelected);
  border-radius: 10px;
  color: var(--invertedText);
  cursor: pointer;
  font-weight: bold;
  padding: 5px 20px;
}

.button:disabled {
  opacity: 0.4;
}
