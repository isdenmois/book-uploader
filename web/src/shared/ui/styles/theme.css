button,
input {
  all: unset;
}

button:focus {
  outline: none;
}

body {
  background-color: var(--background);
}

#app {
  max-width: 600px;
  margin: 0 auto;
  height: 100vh;
  overflow: hidden;
  display: flex;
  flex-direction: column;
}

h1 {
  font-size: 24px;
  color: var(--profileText);
}

.book-list > div:not(:first-child) {
  @apply mt-4;
}

.input {
  background: #ffffff;
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.25);
  border-radius: 10px;
  padding-left: 4px;
}
