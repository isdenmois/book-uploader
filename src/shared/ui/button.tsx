import React from 'react'
import { TouchableOpacity } from 'react-native'
import {
  useRestyle,
  spacing,
  border,
  backgroundColor,
  SpacingProps,
  BorderProps,
  BackgroundColorProps,
  VariantProps,
  createRestyleComponent,
  createVariant,
} from '@shopify/restyle'

import { Box, Theme, Text } from './theme'
import { ActivityIndicator } from './activity-indicator'

const buttonVariant = createVariant({ themeKey: 'buttonVariants' })
const ButtonContainer = createRestyleComponent<
  VariantProps<Theme, 'buttonVariants'> & React.ComponentProps<typeof Box>,
  Theme
>([buttonVariant], Box)

const restyleFunctions = [buttonVariant as any, spacing, border, backgroundColor]

type Props = SpacingProps<Theme> &
  VariantProps<Theme, 'buttonVariants'> &
  BorderProps<Theme> &
  BackgroundColorProps<Theme> & {
    onPress?: () => void
    label?: string
    outline?: boolean
    loading?: boolean
    disabled?: boolean
  }

export const Button = ({ onPress, label, loading = false, disabled = false, variant = 'primary', ...rest }: Props) => {
  const props = useRestyle(restyleFunctions, { ...rest, variant })
  const textVariant = 'button_' + variant

  return (
    <TouchableOpacity disabled={disabled || loading} onPress={onPress}>
      <ButtonContainer
        flexDirection='row'
        borderRadius={44}
        paddingVertical='1'
        paddingHorizontal='4'
        justifyContent='center'
        {...props}
      >
        {loading ? (
          <ActivityIndicator size='small' color={variant === 'secondary' ? 'primary' : 'secondary'} />
        ) : (
          <Text variant={textVariant as Partial<keyof Theme['textVariants']>}>{label}</Text>
        )}
      </ButtonContainer>
    </TouchableOpacity>
  )
}
