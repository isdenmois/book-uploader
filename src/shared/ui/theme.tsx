import React, { FC } from 'react'
import { TouchableOpacity, TouchableOpacityProps, useColorScheme } from 'react-native'
import {
  ThemeProvider as ReThemeProvider,
  createBox,
  createText,
  createTheme,
  useTheme as useThemeRN,
} from '@shopify/restyle'

import * as Colors from './colors'

export const theme = createTheme({
  spacing: {
    0.25: 2,
    0.5: 4,
    1: 8,
    2: 16,
    3: 24,
    4: 32,
    5: 40,
    6: 48,
  },
  colors: Colors.light,
  breakpoints: {},
  textVariants: {
    defaults: {
      fontSize: 16,
    },
    chip: {
      fontSize: 14,
      lineHeight: 18,
    },
    button_primary: {
      fontSize: 16,
      lineHeight: 22,
      color: 'invertedText',
    },
  },
  buttonVariants: {
    primary: {
      backgroundColor: 'profileSelected',
    },
  },
})

export type Theme = typeof theme

const darkTheme: Theme = {
  ...theme,
  colors: {
    ...theme.colors,
    ...Colors.dark,
  },
}

export const Box = createBox<Theme>()
export const Text = createText<Theme>()
export const TouchableBox = createBox<Theme, TouchableOpacityProps & { children: any }>(TouchableOpacity)

export const ThemeProvider: FC = ({ children }) => {
  const currentMode = useColorScheme()

  return <ReThemeProvider theme={currentMode === 'dark' ? darkTheme : theme}>{children}</ReThemeProvider>
}

export const useTheme = () => useThemeRN<Theme>()
