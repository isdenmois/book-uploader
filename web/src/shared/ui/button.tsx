import { Component } from 'solid-js'
import './styles/button.css'

interface Props {
  disabled?: boolean
  type?: 'submit' | 'reset' | 'button'
}

export const Button: Component<Props> = props => {
  return (
    <button class='button' disabled={props.disabled} type={props.type} role='button'>
      {props.children}
    </button>
  )
}
