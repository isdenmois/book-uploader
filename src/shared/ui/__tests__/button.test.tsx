import React from 'react'
import { act, render } from '@testing-library/react-native'
import { ThemeProvider } from '../theme'
import { Button } from '../button'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

test('Button', async () => {
  const { toJSON, rerender, unmount } = render(<Button label='Hello' />, { wrapper: ThemeProvider })

  expect(toJSON()).toMatchSnapshot()

  rerender(<Button label='Hello' loading />)
  expect(toJSON()).toMatchSnapshot()

  await act(async () => {
    unmount()
  })
})
