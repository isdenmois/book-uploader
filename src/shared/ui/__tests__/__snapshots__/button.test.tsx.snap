// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 44,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 32,
          "paddingVertical": 8,
        },
        Array [
          Object {},
          Array [
            Object {
              "backgroundColor": "#FF6842",
            },
          ],
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "display": "flex",
            "fontSize": 16,
            "lineHeight": 22,
          },
        ]
      }
    >
      Hello
    </Text>
  </View>
</View>
`;

exports[`Button 2`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 44,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 32,
          "paddingVertical": 8,
        },
        Array [
          Object {},
          Array [
            Object {
              "backgroundColor": "#FF6842",
            },
          ],
        ],
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#787D87"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;
